### ISSUE TITLE
[Dev] Settings - refactor to using enums

### ISSUE THREAD

**Issue by ljacqu** [Xephi/AuthMeReloaded#312]
Similar to the changes introduced with the permission nodes and message keys, I could imagine there being one or more enums to refer to settings, rather than accessing the instance fields directly. 

I believe this could be done rather generically, defining an enum for the setting type (is it a string, an int?) and then iterating over some SettingKey enum we could choose the right way to load it. An enum field could even hold the default value, and if some planning is involved before refactoring I'm certain we could even pack those special settings that have custom behavior.

It might make sense to first copy the Settings class on a separate branch and to make a small test if a certain design works.

**Comment by Xephi** [comment]
don't do it before we split the config file ;)

**Comment by ljacqu** [comment]
Issue blocked by #347

**Comment by Xephi** [comment]
I close this issue, keep track of all Settings change into this one : 
https://github.com/Xephi/AuthMeReloaded/issues/347

**Comment by ljacqu** [comment]
@Xephi Good idea, the reason I made a separate issue was that I could do it once independently of the rest. I could just leave it on a branch then or something
